action/protocol/staking/protocol.go

```diff
@@ -227,39 +232,61 @@ func (p *Protocol) Handle(ctx context.Context, act action.Action, sm protocol.St
 }
 
 func (p *Protocol) handle(ctx context.Context, act action.Action, csm CandidateStateManager) (*action.Receipt, error) {
-	var topics action.Topics
-	var err error
+	var (
+		topics         action.Topics
+		preFairbankLog *action.Log
+		err            error
+	)
 
 	switch act := act.(type) {
 	case *action.CreateStake:
-		topics, err = p.handleCreateStake(ctx, act, csm)
+		topics, preFairbankLog, err = p.handleCreateStake(ctx, act, csm)
 	case *action.Unstake:
-		topics, err = p.handleUnstake(ctx, act, csm)
+		topics, preFairbankLog, err = p.handleUnstake(ctx, act, csm)
 	case *action.WithdrawStake:
-		topics, err = p.handleWithdrawStake(ctx, act, csm)
+		topics, preFairbankLog, err = p.handleWithdrawStake(ctx, act, csm)
 	case *action.ChangeCandidate:
-		topics, err = p.handleChangeCandidate(ctx, act, csm)
+		topics, preFairbankLog, err = p.handleChangeCandidate(ctx, act, csm)
 	case *action.TransferStake:
-		topics, err = p.handleTransferStake(ctx, act, csm)
+		topics, preFairbankLog, err = p.handleTransferStake(ctx, act, csm)
 	case *action.DepositToStake:
-		topics, err = p.handleDepositToStake(ctx, act, csm)
+		topics, preFairbankLog, err = p.handleDepositToStake(ctx, act, csm)
 	case *action.Restake:
-		topics, err = p.handleRestake(ctx, act, csm)
+		topics, preFairbankLog, err = p.handleRestake(ctx, act, csm)
 	case *action.CandidateRegister:
-		topics, err = p.handleCandidateRegister(ctx, act, csm)
+		topics, preFairbankLog, err = p.handleCandidateRegister(ctx, act, csm)
 	case *action.CandidateUpdate:
-		topics, err = p.handleCandidateUpdate(ctx, act, csm)
+		topics, preFairbankLog, err = p.handleCandidateUpdate(ctx, act, csm)
 	default:
 		return nil, nil
 	}
 
+	blkCtx := protocol.MustGetBlockCtx(ctx)
+	postFairbankMigration := true
+	if p.hu.FairbankBlockHeight() >= 12*720 {
+		postFairbankMigration = blkCtx.BlockHeight >= p.hu.FairbankBlockHeight()-12*720
+	}
```

Comment:
here set migration time = 12 hours before actual fairbank block time

so all the candidates/buckets we import during migration will have correct topics in the log

wanted to know what you think about this, or maybe 8 hours is good enough